lib/scheduler.ml

```diff
@@ -200,15 +165,55 @@ let rec main_loop s =
       Mtime.Span.(abs_diff sleep_until this |> to_s)
     else 0.
   in
-  let (_: bool) = Delay.wait s.delay seconds in
+  let (_: bool) = PipeDelay.wait s.delay seconds in
   main_loop s
 
 let make () =
   let s = {
     schedule = HandleMap.empty;
-    delay = Delay.make ();
+    delay = PipeDelay.make ();
     next_id = 0;
     m = Mutex.create ();
   } in
   let _:Thread.t = Thread.create main_loop s in
   s
+
+module Delay = struct
+  type state = Signalled | Timedout
+  let s = make ()
```

Comment:
i'd rename this `make_scheduler to contract between `Delay.make`

Reply:
or maybe even not have a `make_scheduler` function, since we only make one of them

Reply:
We do create more than one Scheduler. We only create one pipe/scheduler though.
I've added a `make_scheduler` to make it clearer which `make` we are calling.
I retained the toplevel `make` though because that is exported in the mli and used elsewhere.